### PR TITLE
add custom schemas support, expand README.md, move to 18 node, remove…

### PR DESCRIPTION
**Pull request by poputu**
… dependencies

### CODE REVIEW

**Review by hmvs — APPROVED** [review]

**Comment by koresar** [comment]
Can't publish yet as I can't understand what SerVer release it is - MAJOR or MINOR.

This PR breaks the existing API in a few ways. I've merging for now to a new branch `drop_printf_and_add_custom_txs`. Let's discuss why the API changes are necessary.

File: index.js [hunk 1]
```diff
@@ -91,19 +162,22 @@ class ABA {
     formatHeader() {
         const time = new Date(this.options.time || this.options.date || new Date());
 
-        return printf(HEADER_FORMAT, {
-            ...this.options,
+        const headerData = {
+            ...this.options, // breaking change this.options.header
```
Comment (koresar) [inline index.js:166]: What is this comment?

File: index.js [hunk 2]
```diff
@@ -91,19 +162,22 @@ class ABA {
...
             date: pad2(time.getDate()) + pad2(time.getMonth() + 1) + pad2(time.getFullYear() % 100), // DDMMYY
-            bsb: formatBsb(this.options.bsb),
+            bsb: this.options.bsb,
```
Comment (koresar) [inline index.js:168]: This is a breaking change.

File: index.js [hunk 3]
```diff
@@ -65,24 +98,62 @@ class ABA {
      * @param options.description {String} Description of this file entries. Up to 12 chars.
      * @param [options.date=Date.now()] {Date|String|Number} Date to be processed.
      * @param [options.time] {Date|String|Number} Time to be processed. Should be ignored according to the specs.
+     * @param [options.schemas] {Object} Custom schemas
+     * @param [options.customHeaderData] {Object} Custom data for custom header, warning, will overwrite standard field if name will be the same
```
Comment (koresar) [inline index.js:102]: I find this `custom...` option too difficult to understand. Let's rethink. This should be more user friendly.

**Pull request merged**